network_diffusion/mln/mlnetwork_torch.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) 2024 by Michał Czuba.
+#
+# This file is a part of Network Diffusion.
+#
+# Network Diffusion is licensed under the MIT License. You may obtain a copy
+# of the License at https://opensource.org/licenses/MIT
+# =============================================================================
+
+"""A converter from `MultilayerNetwork` to the sparse repr. in PyTorch."""
+
+from dataclasses import dataclass
+from typing import Any
+
+import networkx as nx
+import torch
+from bidict import bidict
+
+from network_diffusion.mln.mlnetwork import MultilayerNetwork
+
+
+def _prepare_mln_for_conversion(
+    net: MultilayerNetwork,
+) -> tuple[MultilayerNetwork, bidict, dict[str, set[Any]] | None]:
+    """
+    Prepare `MultilayerNetwork` for conversion to torch representation.
+
+    If network is not multiplex, then multiplicity actoss all layers will be
+    imposed. Names of the actors will be converted to integers.
+
+    :param net: a copy of the multilayer network prepared for conversion
+    :return: a new instance of `MultilayerNetwork` prepared for conversion,
+        a bi-directional map of the old and new actors' names, a dict of nodes'
+        sets added to make the network multiplex (with their original ids)
+    """
+    if not net.is_multiplex():
+        net_m, added_nodes = net.to_multiplex()
+    else:
+        net_m, added_nodes = net, None
+    ac_map = {ac.actor_id: idx for idx, ac in enumerate(net_m.get_actors())}
+    l_dict = {
+        l_name: nx.relabel_nodes(l_graph, mapping=ac_map, copy=True)
+        for l_name, l_graph in net_m.layers.items()
+    }
+    return MultilayerNetwork(l_dict), bidict(ac_map), added_nodes
+
+
+def _from_scipy_sparse(mat: Any) -> tuple[torch.Tensor, torch.Tensor]:
```

Comment:
Why Any?

Reply:
That's because `scipy` has many implementations of sparse matrices, and I'm not sure which one are we using. Moreover, I've just copied this function from `pyg`, and I thought that I'll not wise to embellish it ;)